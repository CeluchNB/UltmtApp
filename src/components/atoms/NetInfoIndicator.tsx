import { useColors } from '../../hooks'
import { useNetInfo } from '@react-native-community/netinfo'
import { IconButton, Tooltip } from 'react-native-paper'
import React, { useEffect, useState } from 'react'
import { StyleSheet, View } from 'react-native'

const NetInfoIndicator: React.FC<{}> = () => {
    const { colors } = useColors()
    const netInfo = useNetInfo()
    const [iconName, setIconName] = useState('network-strength-3')
    const [iconColor, setIconColor] = useState(colors.success)
    const [tooltipText, setTooltipText] = useState('Good Internet connection')

    useEffect(() => {
        if (!netInfo.isInternetReachable) {
            setIconColor(colors.error)
            setIconName('network-strength-1')
            setTooltipText('Poor Internet connection. Consider offline game.')
        } else {
            setIconColor(colors.success)
            setIconName('network-strength-3')
            setTooltipText('Good Internet connection')
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [netInfo])

    const styles = StyleSheet.create({
        container: { display: 'flex', flexDirection: 'row' },
        button: { margin: 0 },
    })

    return (
        <View style={styles.container}>
            <IconButton
                style={styles.button}
                iconColor={iconColor}
                icon={iconName}
                size={20}
            />
            <Tooltip title={tooltipText} enterTouchDelay={200}>
                <IconButton
                    style={styles.button}
                    iconColor={colors.textPrimary}
                    icon="help-circle"
                    size={20}
                    onPress={() => {}}
                />
            </Tooltip>
        </View>
    )
}

export default NetInfoIndicator
