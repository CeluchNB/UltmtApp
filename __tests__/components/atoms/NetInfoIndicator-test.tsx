import NetInfoIndicator from '../../../src/components/atoms/NetInfoIndicator'
import React from 'react'
import { render } from '@testing-library/react-native'

jest.mock('@react-native-community/netinfo', () => {
    return {
        useNetInfo: () => {
            return {
                isInternetReachable: true,
            }
        },
    }
})
jest.mock('react-native-paper', () => {
    const RNPaper = jest.requireActual('react-native-paper')
    return {
        ...RNPaper,
        Tooltip: () => {
            return <div>Network status indicator</div>
        },
    }
})

describe('NetInfoIndicator', () => {
    it('with reachable internet', async () => {
        const snapshot = render(<NetInfoIndicator />)
        expect(snapshot).toMatchSnapshot()
    })
})
