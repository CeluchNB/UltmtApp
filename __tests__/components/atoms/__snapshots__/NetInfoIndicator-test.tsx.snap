// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetInfoIndicator with reachable internet 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 18,
        "height": 36,
        "margin": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 36,
      }
    }
    testID="icon-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 18,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="icon-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 18,
              },
            ],
          ]
        }
        testID="icon-button"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#40dd50",
                "fontSize": 20,
              },
              [
                {
                  "lineHeight": 20,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ó°£¸
        </Text>
      </View>
    </View>
  </View>
  <div>
    Network status indicator
  </div>
</View>
`;
